pytorch_lightning/loggers/csv_logs.py

```diff
@@ -197,6 +200,8 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
     def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
         metrics = self._add_prefix(metrics)
         self.experiment.log_metrics(metrics, step)
+        if step is not None and (step + 1) % self._flush_logs_every_n_steps == 0:
+            self.save()
```

Comment:
if `step is None` then we'll never trigger saving? 
should we track how many times `log_metrics` is called with a separate counter, and then use that for determining when to save?

Reply:
Right, saving would only be done at the end when `finalize` is being called. I'm not sure about the semantics of having an internal counter, since it would be different from the `global_step` of the trainer. What do you think?